src/categorical_algebra/CSets.jl

```diff
@@ -1621,6 +1621,16 @@ function Base.push!(S::SubobjectIteratorState,h::ACSetTransformation)
   push!(S.seen, h); push!(S.to_recurse, h); push!(S.to_yield, h)
 end
 
+"""This would be much more efficient with canonical isomorph"""
+function seen(S::SubobjectIteratorState, f::ACSetTransformation)
```

Comment:
A small thing, but I like the Julia convention of prefacing the names of predicates with `is` or `has`, so this might be `is_seen`.